Изменения крематория и морга

zipliks opened the pull request:
## What Does This PR Do
Крематорий сжигает все предметы внутри, не выплёвывая наружу.
Изменён спрайт крематория.
Добавлены span классы, чтобы не срать в дальнейший код `<span></span>`

## Why It's Good For The Game
Не нужно класть предметы в коробку, чтобы сжечь в печке, они вполне горят вместе с трупом и мешком для трупов.
Кучки пепла не складируются рядом с крематорием.
Новый спрайт - это стильно, модно, молодёжно.

## Changelog
:cl: Luger
imageadd: Новый спрайт крематория и морга
fix: Крематорий больше не выплёвывает айтемы наружу
tweak: Кучки пепла не спавнятся, если рядом есть пепел
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

zipliks committed:
Fix: fixed crematorium didn't delete items inside

zipliks committed:
Изменение спрайта крематория, фикс сжигания предметов, фикс кучек пепла

zipliks committed:
Инклуды новых файлов

zipliks committed:
Исправлены кривые спрайты, добавлены вариации по сторонам света

Gaxeer commented on code/__DEFINES/span.dm:
```diff
@@ -0,0 +1,4 @@
+#define span_hear(str) ("<span class='hear'>" + str + "</span>")
```
насколько я знаю, дефайны принятно писать капсом SPAN_HEAR, ну или же можно просто HEAR, WARNING, ALERT. Но со спаном думаю лучше. SPAN_HEAR, SPAN_WARNING, SPAN_ALERT

zipliks replied:
Справебыдло, прост спаны в голове не как дефайны ощущались

Gaxeer commented on code/game/atoms.dm:
```diff
@@ -28,6 +28,9 @@
 	var/explosion_block = 0
+	
+	///First atom flags var
+	var/flags_1 = NONE
```
почему _1?

zipliks replied:
var/flags = NONE уже есть
Не уверен создаст ли это излишние конфликты или UB moments

Gaxeer commented on code/game/objects/structures/morgue.dm:
```diff
@@ -41,19 +45,20 @@
-			var/obj/structure/closet/body_bag/B = locate() in contents
-			if(M==null) M = locate() in B
+			var/obj/structure/closet/body_bag/item = locate() in contents
+			if(entity == null)
```
if (!entity)

zipliks replied:
Строго говоря, я занимался только крематорием, весь остальной код меня мало интересовал. Поверхностый рефакторинг произошёл по причине "сука, как это читать".

Gaxeer commented on code/game/objects/structures/morgue.dm:
```diff
@@ -41,19 +45,20 @@
 
-			if(M)
-				var/mob/dead/observer/G = M.get_ghost()
+			if(entity)
```
можно сделать else блок после

zipliks replied:
Причина та же, я не трогал эти куски кода, но могу и потрогать

Vallat replied:
Нельзя тут else блок использовать

Gaxeer commented on code/game/objects/structures/morgue.dm:
```diff
@@ -481,9 +503,9 @@
 	if(stat == DEAD)
 		var/obj/structure/morgue/morgue
 		var/mob/living/C = src
```
я думаю стоит C заменить на creature? Или что тут имелось ввиду

zipliks committed:
Дефайны в SNAKE_UPPER_CASE, заглушый надоедливый ворнинг в `buildandrepair.dm`, небольшие изменения проверок и нейминга

Vallat commented on code/game/atoms.dm:
```diff
@@ -28,6 +28,9 @@
 	var/explosion_block = 0
+	
+	///First atom flags var
+	var/flags_1 = NONE
```
Удалить, ты его нигде не используешь

Vallat commented on code/__HELPERS/atoms.dm:
```diff
@@ -0,0 +1,9 @@
+	var/idx = 0
+	while(idx < length(.))
+		var/atom/checked_atom = .[++idx]
+		if(checked_atom.flags_1 & ignore_flag_1)
```
Обычный flags используй, либо удаляй ваще проверку на игнор флага. Не используется нигде

Vallat commented on code/game/objects/structures/morgue.dm:
```diff
@@ -247,24 +252,29 @@
+	else if(cremating)
+		icon_state = "crema_cremating"
+		return
+	else if(length(contents))
+		icon_state = "crema_active"
+		return
 	else
-		if(contents.len)
-			icon_state = "crema2"
-		else
-			icon_state = "crema1"
-	return
+		icon_state = "crema_default"
+		return
```
Насрано ретёрнами, они бесполезны, убери

zipliks committed:
fix: null safety, runtime error

zipliks committed:
fix: removed useless var

zipliks committed:
feat: added new proc to get specific type from contents

zipliks committed:
refactor: removed useless if-scopes, +icons

Код для морга переписан подстать крематорию
Добавлены спрайты на 4 направления
Спрайты переименованы для лучшей читаемости
Убраны бесполезные return'ы у крематория

zipliks committed:
feat: new icons for morgue + rename

Gaxeer commented on code/game/objects/structures/morgue.dm:
```diff
@@ -35,43 +39,42 @@
 	update()
 
 /obj/structure/morgue/proc/update()
+	var/list/morgue_content     = get_all_contents() - src - connected
+	var/list/morgue_mob         = get_all_contents_type(/mob/living)
```
```suggestion
	var/list/morgue_content = get_all_contents() - src - connected
	var/list/morgue_mob = get_all_contents_type(/mob/living)
```

zipliks committed:
убираем ужасные пробелы

Co-authored-by: Gaxeer <44334376+Gaxeer@users.noreply.github.com>
